test/smtp.test.lua

```diff
@@ -146,6 +150,11 @@ test:test("smtp.client", function(test)
                   m.text,
                   "Subject: =%?utf%-8%?b%?YWJjZGVmZ2hpamvRj2xtbm9wcXJzdHV2d3h5eg==%?=", ""))
     test:is(subj, 1, 'subject codes >127')
+
+    r = client:request(addr, 'sender@tarantool.org',
+                       'unexisting@tarantool.org',
+                       'mail.body')
+    test:is(r.reason, 'SMTP error: RCPT failed: 510', 'unexisting recipient')
```

Comment:
How about 3xx and 4xx responses? How about malformed responses? Those cases are mentioned in the issue.